Lib/shutil.py

```diff
@@ -170,6 +171,35 @@ def _copyxattr(src, dst, *, follow_symlinks=True):
     def _copyxattr(*args, **kwargs):
         pass
 
+if sys.platform == 'darwin':
+    # As of macOS 10.10, the `copyfile` API changed to not copy certain flags
+    mac_ver = tuple(int(x) for x in platform.mac_ver()[0].split('.'))
+    if mac_ver >= (10, 10, 0):
+        def _fix_flags(dst, flags, follow_symlinks):
+            """Perform any modifications to `flags` before they can be applied
+            to `dst` with `chflags()`.
+
+            """
+
+            # From xnu's bsd/sys/stat.h
+            UF_TRACKED    = 0x00000040
+            SF_RESTRICTED = 0x00080000
+
+            # Based on copyfile's copyfile_stat()
+            omit_flags = (UF_TRACKED | SF_RESTRICTED)
+            add_flags  = 0
+
+            # If the kernel automatically put SF_RESTRICTED on the destination
+            # already, we don't want to clear it
```

Comment:
I'm not sure if this is needed, will macOS make files restricted when copying into special locations? The SIP feature only protects selected system files. Without this code the signature for this helper function could be simplified.

Reply:
The logic is the same as Apple's code, so I think it's better to keep it.